ros2action/ros2action/verb/echo.py

```diff
@@ -0,0 +1,292 @@
+from enum import Enum
+import queue
+import sys
+import threading
+
+import rclpy
+from rclpy.qos import qos_profile_action_status_default
+from rclpy.qos import qos_profile_services_default
+from rclpy.qos import QoSProfile
+from rclpy.subscription import Subscription
+from rclpy.type_support import EventMessage
+from rclpy.type_support import MsgT
+
+from ros2action.api import action_name_completer
+from ros2action.api import action_type_completer
+from ros2action.api import get_action_class
+from ros2action.verb import VerbExtension
+
+from ros2cli.helpers import unsigned_int
+from ros2cli.node.strategy import NodeStrategy
+
+from rosidl_runtime_py import message_to_csv
+from rosidl_runtime_py import message_to_ordereddict
+from rosidl_runtime_py.utilities import get_action
+
+from service_msgs.msg import ServiceEventInfo
+
+import yaml
+
+DEFAULT_TRUNCATE_LENGTH = 128
+
+
+class ActionInterfaces(Enum):
+    GOAL_SERVICE = 'GOAL_SERVICE'
+    CANCEL_SERVICE = 'CANCEL_SERVICE'
+    RESULT_SERVICE = 'RESULT_SERVICE'
+    FEEDBACK_TOPIC = 'FEEDBACK_TOPIC'
+    STATUS_TOPIC = 'STATUS_TOPIC'
+
+
+class EchoVerb(VerbExtension):
+    """Echo a action."""
+
+    # Custom representer for getting clean YAML output that preserves the order in an OrderedDict.
+    # Inspired by: http://stackoverflow.com/a/16782282/7169408
+    def __represent_ordereddict(self, dumper, data):
+        items = []
+        for k, v in data.items():
+            items.append((dumper.represent_data(k), dumper.represent_data(v)))
+        return yaml.nodes.MappingNode(u'tag:yaml.org,2002:map', items)
+
+    def __init__(self):
+        self._event_number_to_name = {}
+        for k, v in ServiceEventInfo._Metaclass_ServiceEventInfo__constants.items():
+            self._event_number_to_name[v] = k
+
+        yaml.add_representer(OrderedDict, self.__represent_ordereddict)
+
+    def add_arguments(self, parser, cli_name):
+        arg = parser.add_argument(
+            'action_name',
+            help="Name of the ROS action to echo (e.g. '/fibonacci')")
+        arg.completer = action_name_completer
+        arg = parser.add_argument(
+            'action_type', nargs='?',
+            help="Type of the ROS action (e.g. 'example_interfaces/action/Fibonacci')")
+        arg.completer = action_type_completer
+        parser.add_argument(
+            '--interfaces', '-i', type=str, metavar='interfaces_name_list',
+            help='Specify a action interfaces list are output. "|" is used as a delimiter for '
+                 'multiple action interfaces. Action interfaces include "GOAL_SERVICE", '
+                 '"CANCEL_SERVICE", "RESULT_SERVICE", "FEEDBACK_TOPIC" and "STATUS_TOPIC". '
+                 'If this option is not set, output messages from all interfaces of the action.')
+        parser.add_argument(
+            '--queue-size', '-q', type=unsigned_int, default=100,
+            help='The length of output message queue. The default is 100.')
```

Comment:
I think it is unclear for user what this is for? could we add more descriptions for this queue size optional argument? IMO this does not need to be exposed by user to avoid the complication, but set enough queue length by system?

Reply:
Okay, I will modify the description. 
It is difficult to set it automatically because we don't know how many action messages will arrive simultaneously.

Reply:
Remove this argument.